### PR TITLE
HTML editor

### PR DESCRIPTION
**Pull request by nathanstitt**
image

### CODE REVIEW

**Comment by nathanstitt** [comment]
@camliu89 I pulled in master to get the lint fix PR

File: tutor/src/components/editor/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    if (html) {
+      setHTML(html);
+      setEditing(false);
+    }
+  }, [setEditing, setHTML]);
+  const body = isEditing ?
+    <Editing html={currentHTML} onSave={onSave} /> : <HTML html={currentHTML} onClick={() => setEditing(true)}/>;
+
+  return (
+    <div style={{ margin: '100px', height: '500px' }}>
```
Comment (camliu89) [inline tutor/src/components/editor/index.js:38]: maybe use `styled` here? or are you overriding some parent styles?

**Comment by camliu89** [comment]
@nathanstitt it looks good. I am not sure about the `Save` button in the editor toolbar. Do the users need to save the question text before they can save the question?

**Review by camliu89 — APPROVED** [review]
looks good @nathanstitt , just left a couple of comments.

File: tutor/src/components/editor/index.js (hunk 1 above)
Comment (nathanstitt) [reply tutor/src/components/editor/index.js:38]: good idea, that was just being lazy and copying from the demo which didn't have styled-components

**Comment by nathanstitt** [comment]
> @nathanstitt it looks good. I am not sure about the `Save` button in the editor toolbar. Do the users need to save the question text before they can save the question?

No but it needs a way to switch from editing html to displaying it.  Perhaps the "Save" button isn't really the best control, but  I'm not sure what'd be better

**Comment by camliu89** [comment]
> > @nathanstitt it looks good. I am not sure about the `Save` button in the editor toolbar. Do the users need to save the question text before they can save the question?
> 
> No but it needs a way to switch from editing html to displaying it. Perhaps the "Save" button isn't really the best control, but I'm not sure what'd be better

Maybe just an `onBlur` and `onFocus` listener?

**Pull request merged**